components/pkg-export-kubernetes/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate clap;
+extern crate env_logger;
+extern crate habitat_core as hcore;
+extern crate habitat_common as common;
+#[macro_use]
+extern crate log;
+
+use clap::App;
+use std::result;
+use std::str::FromStr;
+
+use hcore::PROGRAM_NAME;
+use hcore::env as henv;
+use hcore::package::PackageIdent;
+use common::ui::{Coloring, UI, NOCOLORING_ENVVAR, NONINTERACTIVE_ENVVAR};
+
+// Synced with the version of the Habitat operator.
+pub const VERSION: &'static str = "0.1.0";
+
+fn main() {
+    env_logger::init().unwrap();
+    let mut ui = ui();
+    if let Err(e) = start(&mut ui) {
+        let _ = ui.fatal(e);
+        std::process::exit(1)
+    }
+}
+
+fn ui() -> UI {
+    let isatty = if henv::var(NONINTERACTIVE_ENVVAR)
+        .map(|val| val == "true")
+        .unwrap_or(false)
+    {
+        Some(false)
+    } else {
+        None
+    };
+    let coloring = if henv::var(NOCOLORING_ENVVAR)
+        .map(|val| val == "true")
+        .unwrap_or(false)
+    {
+        Coloring::Never
+    } else {
+        Coloring::Auto
+    };
+    UI::default_with(coloring, isatty)
+}
+
+fn start(_ui: &mut UI) -> result::Result<(), String> {
+    let m = cli().get_matches();
+    debug!("clap cli args: {:?}", m);
+    let count = m.value_of("COUNT").unwrap_or("1");
+    // clap_app!() ensures that we do have the mandatory args so unwrap() is fine here
+    let pkg_ident_str = m.value_of("PKG_IDENT").unwrap();
+    let pkg_ident = match PackageIdent::from_str(pkg_ident_str) {
+        Ok(pi) => pi,
+        Err(e) => return Err(format!("{}", e)),
+    };
+    let image = m.value_of("IMAGE").unwrap_or(pkg_ident_str);
```

Comment:
I think this default value will cause issues, since a Habitat identifier isn't always a valid Docker image identifier.

Reply:
So if things remain as they are, this argument should be mandatory?

Reply:
We might construct a value for `IMAGE` from the origin and package name of the given identifier.

Reply:
sure but wouldn't we end up with pkg_ident_str anyway? I think i'll need to change this logic in v2 of this exporter when I add docker image generation.